### ISSUE TITLE
Snapshot webpages longer than 32767 pixels? split into overlapping chunks?

### ISSUE THREAD

**Issue by pstein** [M-Reimer/savescreenshot#26]
Have a look at the following sample web page:

https://www.tenforums.com/tutorials/1950-clean-install-windows-10-a.html

At first scroll down to the bottom of the page to load all pictures with lazy loading.

When I snapshot this page with SaveScreenshot AddOn then only the topmost 32767 pixels are printed.

Maybe there is a format limit for *.png files?

Can you provide a complete snapshot of the full Webpage?

At least a warning should be shown: "Webpage longer that 32767 pixels. Cutting PNG"

As workaround ideas: 

Could you offer an automatic split of the adjacent chunks with an overlapping area of let say 500 pixels?

Furthermore: Paste the next part not into a separate *.png files but at the right side (as a second column).

Limit to 1 additional part would be sufficient. I have never seen webpages longer than 65534 pixels

Thank you
Peter

**Comment by M-Reimer** [comment]
I would say: If you really need longer screenshots, you use the wrong tool for the job. Why don't you save this page as PDF?

The 32767 pixel limit is a hard limit forced by the API, I'm using. If I send values higher than that it just causes an exception and no screenshot happens at all.
But you are right that I could display a notification that the page size got truncated as it exceeded Firefox limits.

But I'll wait with all changes as Mozilla currently reworks some API which is needed for screenshots. We'll see if the new API still has the 32767 pixel limit or if there is a different limit I have to check for.

**Comment by M-Reimer** [comment]
I've checked this with the new API and the same limitations apply.

Just to verify, I did the same test with the built in screenshot utility from Mozilla and it has the same limit at 32767 pixels.

I still say that everyone, who needs such long screenshots, did choose the wrong tool for the task.

I would recommend to get this one: https://addons.mozilla.org/firefox/addon/print-edit-we/

It has a feature to save a website to a PDF file.

I think a warning message wouldn't be a bad idea so I'll keep this one open until it is added.
